<?xml version="1.0"?>
<launch>
  <!-- Standalone example launch file -->

  <!-- Declare arguments with default values -->
  <arg name="port" default="/dev/ttyS1" />
  <arg name="baudrate" default="115200" />
  <arg name="imu_rate" default="10" />
  <arg name="imu_frame_id" default="base_link" />
  <arg name="debug" default="false" />
  <arg name="diagnostics" default="false" />

  <!-- For setting debug level to debug -->
  <group if="$(arg debug)">
    <env name="ROSCONSOLE_CONFIG_FILE" 
       value="$(find microstrain_3dm_gx5_45)/config/custom_rosconsole.conf"/>
  </group>

  <!-- Microstrain sensor node -->
  <node name="microstrain_3dm_gx5_45_node" 
  pkg="microstrain_3dm_gx5_45" 
  type="microstrain_3dm_gx5_45_node" output="screen" respawn="true">
    <!--<param name="port" value="/dev/ttyACM0" type="str" />-->
    <param name="port" value="$(arg port)" type="str" />
    <param name="baudrate" value="$(arg baudrate)" type="int" />

    <param name="device_setup" value="true" type="bool" />
    <param name="readback_settings" value="true" type="bool" />
    <param name="save_settings" value="true" type="bool" />
    <param name="auto_init" value="true" type="bool" />
    <param name="gps_rate" value="4" type="int" />
    <param name="imu_rate" value="$(arg imu_rate)" type="int" />
    <param name="nav_rate" value="10" type="int" />
    <param name="dynamics_mode" value="1" type="int" />
    <param name="declination_source" value="2" type="int" />
    <param name="declination" value="0.23" type="double" />
    
    <param name="gps_frame_id" value="wgs84" type="str" />
    <param name="imu_frame_id" value="$(arg imu_frame_id)" type="str" />
    <param name="odom_frame_id" value="wgs84" type="str" />
    <param name="odom_child_frame_id" value="base_link" type="str" />

    <param name="publish_gps" value="true" type="bool" />
    <param name="publish_imu" value="true" type="bool" />
    <param name="publish_odom" value="true" type="bool" />
  </node>

  <!-- Diagnostics -->
  <group if="$(arg diagnostics)">
    <node pkg="rqt_topic" type="rqt_topic" name="rqt_topic"/>
    <!--<node pkg="rqt_plot" type="rqt_plot" name="pid_setpoints"
      args="/yaw_pid_debug/Setpoint /vel_pid_debug/Setpoint"/>-->
  </group>
  
</launch>
